zoop_wrapper/models/transaction.py

```diff
@@ -391,3 +395,51 @@ def get_card_present_required_fields(cls):
         """
         fields = cls.get_required_fields()
         return fields.union({"amount", "usage"})
+
+
+class InstallmentPlan(ZoopObject):
+    INTEREST_FREE_MODE = "interest_free"
+    WITH_INTEREST_MODE = "with_interest"
+
+    INSTALLMENT_PLAN_MODES = {INTEREST_FREE_MODE, WITH_INTEREST_MODE}
+
+    @classmethod
+    def get_required_fields(cls):
+        fields = super().get_required_fields()
+        return fields.union({"mode", "number_installments"})
+
+    def validate_custom_fields(self, **kwargs):
+        errors = []
+        if self.mode not in self.INSTALLMENT_PLAN_MODES:
+            errors.append(
+                FieldError(
+                    "mode",
+                    f"O mode é inválido! Deveria ser um dos "
+                    f"dois tipos: {self.INSTALLMENT_PLAN_MODES}",
+                )
+            )
+
+        if InstallmentPlan._validate_number_installments(self.number_installments):
+            errors.append(
+                FieldError(
+                    "number_installments",
+                    f"O number_installments é inválido! Deveria ser de 1 até 12, "
+                    f"e não {self.number_installments}",
+                )
+            )
+
+        return errors
+
+    @classmethod
+    def _validate_number_installments(cls, number_installments):
+        """
+        Esse método verifica se:
+            - number_installments é inteiro
+            - number_installments é um valor inteiro entre 1 e 12 incluindo as bordas
+        :return: bool
+        """
+        return (
+            not isinstance(number_installments, int)
+            or number_installments < 1
+            or number_installments > 12
+        )
```

Comment:
Achei confuso.

Pq vc está retornando True se o `number_installments` não for válido.

Melhor deixar 'semanticamente' correto. Retornar True se for válido e False se for inválido!

Reply:
Done.